src/proto/plumtree.rs

```diff
@@ -206,9 +206,19 @@ impl Gossip {
 #[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
 pub struct IHave {
     /// Id of the message.
-    id: MessageId,
+    pub(crate) id: MessageId,
     /// Delivery round of the message.
-    round: Round,
+    pub(crate) round: Round,
+}
+
+impl IHave {
+    pub(crate) fn postcard_encoded_size() -> usize {
+        postcard::experimental::serialized_size(&Self {
+            id: MessageId([1u8; 32]),
+            round: Round(u16::MAX),
+        })
+        .unwrap()
+    }
```

Comment:
it would be great if this was a constant. `postcard::experimental` also as a derive `MaxSize`. This creates a public `IHave::POSTCARD_MAX_SIZE` coming from a `MaxSize` trait. `IHave` is not publicly accesible so this might be a better option

Reply:
TIL! That's much better.